crates/red_knot_python_semantic/resources/mdtest/narrow/match.md

```diff
@@ -64,3 +64,53 @@ match x:
 
 reveal_type(x)  # revealed: object
 ```
+
+## Value patterns
+
+```py
+def get_object() -> object:
+    return object()
+
+x = get_object()
+
+reveal_type(x)  # revealed: object
+
+match x:
+    case "foo":
+        reveal_type(x)  # revealed: Literal["foo"]
+    case 42:
+        reveal_type(x)  # revealed: Literal[42]
+    case 6.0:
+        reveal_type(x)  # revealed: float
+    case 1 + 1j:
+        reveal_type(x)  # revealed: int | float | complex
```

Comment:
nit: I'd suggest for clarity we use a pure complex (i.e. imaginary) literal here. I spent a while wondering why we get `int  | float | complex` here, and it took me a while to realize that it's because `1+1j` is not really a literal, its a binop Add with int and complex operands.
```suggestion
    case 1j:
        reveal_type(x)  # revealed: complex
```